### ISSUE TITLE
Drop Emacs 24.x compatibility

### ISSUE THREAD

**Issue by swsnr** [flycheck/flycheck#623]
Remove all compatibility shims for 24.x releases of Emacs.

**Comment by Sarcasm** [comment]
I don't think it should be a *goal* to remove backward compatibility in itself. It should be something that happens when one has to make too many compromise to support an old version, used by a minority.

You are providing a tool useful to many users and that is used by third party plugins. Not supporting the Emacs 24.x series mean third party tools will also have to drop support for it.

I guess you don't plan on doing that just now but I'm surprised to see an issue that specifically state that you "have" to remove support for Emacs 24.x.

**Comment by lunaryorn** [comment]
@Sarcasm Oh dear, calm down.  Emacs 25 isn't anywhere near a release yet…

**Comment by porglezomp** [comment]
I agree that this is bizarre as a goal. It seems like you shouldn't be dropping 24.x support until Emacs 26 is on the horizon.

**Comment by cpitclaudel** [comment]
@porglezomp Our goal is not to drop support for Emacs 24 any time soon; no worries. 

This item is just a gentle reminder to the devs that the code has a number of compatibility shims for features that Emacs 25 supports natively, which we'll be able to remove when we make 25.0 the required version.

It's fun that your message came just after Emacs 25.0's feature freeze, where it was suggested to [bump master to 26.0](https://lists.gnu.org/archive/html/emacs-devel/2015-11/msg01357.html); maybe "until Emacs 26 is on the horizon" is actually pretty soon :)

**Comment by lunaryorn** [comment]
Adding to @cpitclaudel, I'd like to note that we will drop Emacs 24 compatibility at some point after Emacs 25 is released.  We will very likely not wait for Emacs 26, for that could be years from now—just take a look at when Emacs 24.1 was released.  But we have made no choice yet when we'll drop Emacs 24 compatibility and we'll consider the needs of our users and downstream distributions before making that choice.

And finally, on a personal note, I'd like to ask you to reconsider your passive-aggressive voice and your demanding tone.  We are willing to discuss this matter with the community but only on a basis of mutual respect and understanding.
